mantidproject/mantid
# Indirect Diffraction - Manual Grouping Bug

**Description of work.**
This PR fixes a bug where the wrong component is passed to the CreateGroupingWorkspace which means the relevant detector IDs couldn't be retrieved from the OSIRIS IPF when using manual grouping on Indirect Diffraction.

**To test:**
1.`Interfaces`->`Indirect`->`Diffraction`
2. Instrument is `OSIRIS` and reflection is `diffspec`
2. Enter the run numbers `137793-137794`
3. Tick manual grouping and enter the number of groups to be 1.
4. Click `Run` and there should be no error.
5. Now untick manual grouping
6. Tick vanadium runs and enter `137713-137714`
7. CLick `Run` and wait.
8. Click `Plot` (under run). The plot for 137793 should look like this:

![image](https://user-images.githubusercontent.com/40830825/53882845-536f4900-400f-11e9-968b-a9f40e83c4b0.png)

Fixes #24974 #25071

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
